lapce-app/src/terminal/data.rs

```diff
@@ -748,17 +748,30 @@ impl TerminalData {
     }
 
     pub fn stop(&self) {
-        if let Some(dap_id) = self.run_debug.with_untracked(|x| {
+        let (dap_id, stopped) = self.run_debug.with_untracked(|x| {
             if let Some(process) = x {
                 if !process.is_prelaunch && process.mode == RunDebugMode::Debug {
-                    return Some(process.config.dap_id);
+                    return (Some(process.config.dap_id), process.stopped);
+                } else {
+                    return (None, process.stopped);
                 }
             }
-            None
-        }) {
-            self.common.proxy.dap_stop(dap_id);
+            (None, true)
+        });
+
+        if !stopped {
+            if let Some(dap_id) = dap_id {
+                self.common.proxy.dap_stop(dap_id);
+            }
+            self.common.proxy.terminal_close(self.term_id);
+            if let Err(err) = self
+                .common
+                .term_tx
+                .send((self.term_id, TermEvent::CloseTerminal))
+            {
+                tracing::error!("{:?}", err);
+            }
         }
-        self.common.proxy.terminal_close(self.term_id);
```

Comment:
What's wrong with the current approach?

Reply:
If the program has already stopped, the original process will repeatedly send these events.Therefore, this time, a check for stopped was added.

Reply:
But even if the program is stopped, `TerminalSender` isn't dropped from `terminals` in `Dispatcher`, so it still needs this event to clean that up.

Reply:
When the program stops, it sends a _CoreNotification::TerminalProcessStopped_, after which the terminals remove the related datas.

![1725611113090](https://github.com/user-attachments/assets/6b7c8803-ef09-4f03-9ad4-0947636a42df)

Reply:
That's the proxy notify the core that the program stopped so that the GUI knows. 

I meant the terminals in `Dispatcher` needs to be cleaned up. 
https://github.com/lapce/lapce/blob/ef65fa614a2ff922244fa14019c867bb5c26fc59/lapce-proxy/src/dispatch.rs#L222

Reply:
You are right. If I add `self.common.proxy.terminal_close(*term_id);`  here(https://github.com/jm-observer/lapce/blob/6e25309f914acc4996b6d9e731a094f34a8eb13f/lapce-app/src/window_tab.rs#L2078), will it be correct?

Reply:
I think it's right as is. 

So terminal stop means the program stopped, for normal terminal, it means it automatically "close" the terminal tab as well. But for run debug, it doesn't close the terminal tab. 

Terminal close means the actual terminal tab is closed.